### ISSUE TITLE
Postgres timestamp precision is always zero with Atlas

### ISSUE THREAD

**Issue by sagikazarmark** [ent/ent#2454]
- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/ent/ent/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯

When a field is defined as `timestamp` (or any other alternatives, like `timestamptz`), the precision in the database type is always 0 (regardless if a precision was specified or not).

## Expected Behavior 🤔

When no precision is specified, the database type shouldn't have a precision.
When a precision is specified, the database type should have that precision, not zero.

## Steps to Reproduce 🕹

Steps:

1. Create a time field
2. Enable atlas
3. Describe the table

```go
var timeSchema = map[string]string{
	dialect.Postgres: "timestamptz",
	dialect.SQLite:   "timestamp",
	dialect.MySQL:    "datetime(3)",
}

// ...

field.Time("expiry").
    SchemaType(timeSchema),
```

## Your Environment 🌎

| Tech        | Version |
| ----------- | ------- |
| Go          | 1.17.3  |
| Ent         | 0.10.1   |
| Database    | Postgres   |
| Driver      | https://github.com/lib/pq        |

I was able to reproduce the issue in this PR: https://github.com/dexidp/dex/pull/2428

One additional detail (that should probably end up in a separate issue): even when Atlas is disabled, precision values are ignored. Even if I set the type to `timestamptz(3)` for example, in the database the type will just be `timestamp with time zone`. This is fine in my case because the default is not zero fortunately.

**Comment by a8m** [comment]
Thanks for reporting this @sagikazarmark, I'll give it a look now.

**Comment by masseelch** [comment]
For reference: https://github.com/ariga/atlas/pull/686 and https://github.com/ariga/atlas/pull/687.

**Comment by a8m** [comment]
I'll update the repository soon with the fix.

**Comment by a8m** [comment]
Fixed with #2462. Thanks, @sagikazarmark 🙏

**Comment by sagikazarmark** [comment]
Awesome, thanks for the quick fix!  🎉
